.github/workflows/ci.yml

```diff
@@ -0,0 +1,21 @@
+name: CI
+on: [push, pull_request]
+jobs:
+  test:
+    name: Node ${{ matrix.node }}, Mongo ${{ matrix.mongodb-version }}, on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node: ['12', '14', '16']
```

Comment:
We need to add node versions down to v4 to guarantee that we're supporting those versions to remain backward-compatible.

Otherwise, some changes might be using a syntax that's not supported in older versions and will break apps out there.

Reply:
I increased in package.json the necessary Version to 12. 

To be Backwards compatible we need to change the lets and consts to var and use the old mocha version. It has perf impacts to use vars. 

But like I said... It is an easy change. 

So do we really need to support from 4 and above? Or can we drop support for non lts versions of node?

Reply:
We _could_ drop the support of the older versions, but we'd need to release a new major version to do that.

This would be fine for mongoose since we only support Node >=12, but if another package out there is relying on _mpath_, that means they won't be able to support any version lower than node 12.

A solution that gets us the best of both worlds would be:
1- Release a minor version with backward-compatible performance improvements (_var_ instead of _let/const_).
2- Release a major version with the least support of Node 12* where we can use _let/const_.

\* Instead of picking an arbitrary version (12), we should support as many versions as we can.
So, for example, if all the optimizations work with node 6, there would be no reason for us to declare version 6 unsupported. 

Using an older mocha should be fine, are there any new features in mocha that we're relying on?

Reply:
I agree.

Reply:
Done... it is now working in node 4 like it is working in node 16